src/encoding/integer/rle_v1.rs

```diff
@@ -147,6 +167,139 @@ impl<N: NInt, R: Read, S: EncodingSign> GenericRle<N> for RleV1Decoder<N, R, S>
     }
 }
 
+pub struct RleV1Encoder<N: NInt, S: EncodingSign> {
```

Comment:
Do you think it's possible to rewrite the encoding algorithm here to rely less on a global mutable state, similar to what was done for RLEv2?

https://github.com/datafusion-contrib/orc-rust/blob/f934c8eab4915836b8655e61bec5ddd959113c50/src/encoding/integer/rle_v2/mod.rs#L226-L366

Would prefer not to have a straight lift and shift from the original ORC code as I don't find it very readable and hard to understand.